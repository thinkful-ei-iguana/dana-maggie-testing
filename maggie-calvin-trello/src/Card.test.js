import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import store from './store';
import renderer from 'react-test-renderer';

describe('<Card />', ()=>{
it('renders without crashing', () => {
  const div = document.createElement('div');
  const title = store.allCards['a'].title;
  const content = store.allCards['a'].content;
  ReactDOM.render(<Card title={title} content={content}/>, div);
  ReactDOM.unmountComponentAtNode(div);

});

it('Renders UI as expected', ()=> {
  const tree = renderer.create(<Card title={"This is a card"}
                                      content={"this is the content of the card"}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
});