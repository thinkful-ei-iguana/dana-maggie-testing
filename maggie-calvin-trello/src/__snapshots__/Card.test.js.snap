// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Renders UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    This is a card
  </h3>
  <p>
    this is the content of the card
  </p>
</div>
`;
